### PR TITLE
Fixes the new inline download not working properly due to missing filename in URL and missing BlobDispatcher handling

### PR DESCRIPTION
**Pull request by mko-sci**
Fixes: [SIRI-749](https://scireum.myjetbrains.com/youtrack/issue/SIRI-749/URLBuilder-Inline-Download-handling-ermoglichen)

### CODE REVIEW

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/biz/storage/layer2/URLBuilder.java [hunk 1]
```diff
@@ -507,6 +505,14 @@ private UrlResult createPhysicalDeliveryUrlResult() {
         return new UrlResult(result.toString(), UrlType.PHYSICAL);
     }
 
+    private void appendFilename(StringBuilder sb, String physicalKey, String filename) {
+        sb.append(physicalKey);
+        sb.append("/");
+        appendAddonText(sb);
+        sb.append(Strings.urlEncode(Files.toSaneFileName(filename)
+                                         .orElse(physicalKey + fetchUrlEncodedFileExtension())));
+    }
```
Comment (sabieber) [inline src/main/java/sirius/biz/storage/layer2/URLBuilder.java:514]:
```suggestion
    private void appendFilename(StringBuilder result, String physicalKey, String filename) {
        result.append(physicalKey);
        result.append("/");
        appendAddonText(result);
        result.append(Strings.urlEncode(Files.toSaneFileName(filename)
                                         .orElse(physicalKey + fetchUrlEncodedFileExtension())));
    }
```

**Review by sabieber — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

File: src/main/java/sirius/biz/storage/layer2/BlobDispatcher.java [hunk 2]
```diff
@@ -181,12 +181,26 @@ public DispatchDecision dispatch(WebContext request) throws Exception {
         Values uriParts = Values.of(uri.split("/"));
         String type = uriParts.at(0).asString();
         if (Strings.areEqual(type, PHYSICAL_DELIVERY) && uriParts.length() == 5) {
-            String filename = stripAdditionalText(uriParts.at(4).asString());
+            String physicalKey = stripAdditionalText(uriParts.at(4).asString());
```
Comment (bkositza) [inline src/main/java/sirius/biz/storage/layer2/BlobDispatcher.java:184]: Would it perhaps be possible to eliminate the MagicNumbers? Wherever uriParts.at() is used a descriptive constant might be helpful.

**Review by bkositza — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/biz/storage/layer2/BlobDispatcher.java [hunk 3]
```diff
@@ -181,12 +181,26 @@ public DispatchDecision dispatch(WebContext request) throws Exception {
         Values uriParts = Values.of(uri.split("/"));
         String type = uriParts.at(0).asString();
         if (Strings.areEqual(type, PHYSICAL_DELIVERY) && uriParts.length() == 5) {
```
Comment (bkositza) [inline src/main/java/sirius/biz/storage/layer2/BlobDispatcher.java:183]: What is the difference between a length of 5 or 6? Constants could also be helpful here.

**Review by bkositza — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/biz/storage/layer2/BlobDispatcher.java (hunk 3 above)
Comment (mko-sci) [reply src/main/java/sirius/biz/storage/layer2/BlobDispatcher.java:183]: Also please read my comment above.
The main difference is the length of the URL or in otherwords the number of parts of which the URL consists of

File: src/main/java/sirius/biz/storage/layer2/BlobDispatcher.java (hunk 2 above)
Comment (bkositza) [reply src/main/java/sirius/biz/storage/layer2/BlobDispatcher.java:184]: Would appreciate it if you create the constants as you described. Only the ones that are really used should probably be enough or?

**Review by bkositza — CHANGES_REQUESTED** [review]
As answered in the comment, I would appreciate it if you create the constants as described by you.

File: src/main/java/sirius/biz/storage/layer2/BlobDispatcher.java (hunk 2 above)
Comment (mko-sci) [reply src/main/java/sirius/biz/storage/layer2/BlobDispatcher.java:184]: I agree

**Review by bkositza — APPROVED** [review]

**Pull request merged**